File: performance-comparison/benchmarks/string_processing/clojure/text_search.clj
(import '[clojure.string :as str])
(let [hs-size (Long/parseLong (first *command-line-args*))
      hay     (apply str (map #(if (< (mod % 10) 5) "x" (str (get "hello" (- (mod % 10) 5))))
                              (range hs-size)))
      needle  "hello"
      nlen    (count needle)
      cnt     (loop [pos 0 c 0]
                (let [i (.indexOf hay needle pos)]
                  (if (< i 0) c
                      (recur (+ i nlen) (inc c)))))]
  (println cnt))

File: performance-comparison/benchmarks/string_processing/python/text_search.py
import sys

hs_size = int(sys.argv[1]) if len(sys.argv) > 1 else 10000
needle = "hello"
hay = "".join("x" if i % 10 < 5 else needle[i % 10 - 5] for i in range(hs_size))
count = 0
pos = 0
while True:
    idx = hay.find(needle, pos)
    if idx < 0:
        break
    count += 1
    pos = idx + len(needle)
print(count)
